### PR TITLE
WIP: Github actions

### PR DESCRIPTION
**Pull request by arokem**
This is what a GA workflow for unit testing might look like. 

We can also implement some of the other jobs/pipelines currently implemented by CircleCI:

- [x] Doc building
- [ ] Deploying to PyPI (need to deal with versioneer)

### CODE REVIEW

**Comment by arokem** [comment]
I think that this is ready to go. This now (hopefully) does: test-suite, documentation build and docker image building. I will do the pypi deploy stuff on a separate PR.

**Comment by arokem** [comment]
Please hold off on merging this for now. The recent 079a3b2729b1f51f694b2ca5798e439d563313a0 tries to use the canned action again. I might back off this in the end, but the developer of this action has made the additional effort of implementing this, so the least I can do is give it a whirl.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install software
+      run: |
+        pip install .[all]
+    - name: Test
+      run: |
+        pytest -n 2 -vv --doctest-modules --cov dmriprep --cov-config .coveragerc --cov-report xml:cov.xml dmriprep
```
Comment (oesteban) [inline .github/workflows/test.yml]: the options to pytest would be better set within the setup.cfg file.

Comment (arokem) [reply .github/workflows/test.yml]: Done in 0cc7787

File: .github/workflows/docbuild.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      with:
+        python-version: '3.7'
+    - name: Install software
+      run: |
+        pip install .[all]
+    - name: Build docs
+      run: |
+        cd docs
+        make html
+    - name: Upload docs
```
[the remark this answers is not in the document]
Comment (arokem) [reply .github/workflows/docbuild.yml]: Done (maybe?) in 313f5bf

**Comment by oesteban** [comment]
I have added some github actions workflow to test package builds. Can we split this one and maybe add those workflows we want to keep?

**Comment by oesteban** [comment]
Closing for now.

**Pull request closed without merging**